File: go-algorithm/leetcode/problem326.go
package leetcode

// 326. Power of Three
// https://leetcode-cn.com/problems/power-of-three/
func isPowerOfThree(n int) bool {
	if n <= 0 {
		return false
	}
	if n == 1 {
		return true
	}
	rem := n % 3
	if rem != 0 {
		return false
	}
	return isPowerOfThree(n / 3)
}

File: python-algo/leetcode/problem_326.py
"""326. Power of Three
https://leetcode.com/problems/power-of-three/
"""


class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        if n == 1:
            return True
        if n < 1:
            return False
        n, m = divmod(n, 3)
        if m != 0:
            return False
        return self.isPowerOfThree(n)
